src/box/sql/expr.c

```diff
@@ -3192,13 +3192,17 @@ expr_code_int(struct Parse *parse, struct Expr *expr, bool is_neg,
 		size_t len = strlen(z);
 		bool unused;
 		if (sql_atoi64(z, &value, &unused, len) != 0 ||
-		    (is_neg && (uint64_t) value > (uint64_t) INT64_MAX + 1)) {
+		    (is_neg && (uint64_t)value > (uint64_t)INT64_MAX + 1)) {
```

Comment:
Since this patch only fixes the mentioned issue, I don't see the need for this refactoring.